libqtile/backend/x11/core.py

```diff
@@ -609,6 +609,9 @@ def handle_KeyPress(self, event) -> None:  # noqa: N802
         # As we're grabbing async we can't just replay it, so...
         # We need to forward the event to the focused window
         if not handled and self.qtile.current_window:
+            # Shush mypy
+            # X11 windows always have a window attribute
+            assert hasattr(self.qtile.current_window, "window")
```

Comment:
I've added this to shush mypy but it still says Window has no attribute window? Any reason why?

Reply:
`hasattr` here will evaluate to `True` even if `self.qtile.current_window.window` is `None`. Maybe you were meaning to use `getattr`? You could do `assert self.qtile.current_window.window`. Or alternatively `if self.qtile.current_window.window:`

Reply:
Oh that makes sense, I will change this

Reply:
Still doesn't work :(, I wanted to do hasattr as window is not defined as a member in the backend window base

Reply:
Maybe `assert getattr(self.qtile.current_window, "window", False)`? Then it's saying that the attribute exists and is truthy

Reply:
Good idea!, trying...

Reply:
Still the same, maybe there is some ignore comment we can do?

Reply:
Maybe `assert isinstance(self.qtile.current_window, (window.Window, window.Static))`? mypy is likely inferring that the window is the base window class. Otherwise `# type: ignore` on the offending lines, presumably where `current_window.window` is accessed.

Reply:
`assert isinstance(self.qtile.current_window, (window.Window, window.Static))` worked, thanks a lot!